packages/opentelemetry-plugin-mysql/src/mysql.ts

```diff
@@ -0,0 +1,266 @@
+import { getConnectionAttributes, getSpanName } from './utils';
+
+export class MysqlPlugin extends BasePlugin<typeof mysqlTypes> {
+  readonly moduleName = 'mysql';
+  readonly supportedVersions = ['2.*'];
+
+  static readonly COMPONENT = 'mysql';
+  static readonly DB_TYPE = 'sql';
+
+  static readonly COMMON_ATTRIBUTES = {
+    [AttributeNames.COMPONENT]: MysqlPlugin.COMPONENT,
+    [AttributeNames.DB_TYPE]: MysqlPlugin.DB_TYPE,
+    [AttributeNames.PEER_SERVICE]: MysqlPlugin.COMPONENT,
+  };
+
+  protected patch(): typeof mysqlTypes {
+    shimmer.wrap(
+      this._moduleExports,
+      'createConnection',
+      this._patchCreateConnection() as any
+    );
+
+    shimmer.wrap(
+      this._moduleExports,
+      'createPool',
+      this._patchCreatePool() as any
+    );
+
+    shimmer.wrap(
+      this._moduleExports,
+      'createPoolCluster',
+      this._patchCreatePoolCluster() as any
+    );
+
+    return this._moduleExports;
+  }
+
+  protected unpatch(): void {
+    shimmer.unwrap(this._moduleExports, 'createConnection');
+    shimmer.unwrap(this._moduleExports, 'createPool');
+    shimmer.unwrap(this._moduleExports, 'createPoolCluster');
+  }
+
+  // global export function
+  private _patchCreateConnection() {
+    return (originalCreateConnection: typeof mysqlTypes.createConnection) => {
+      const thisPlugin = this;
+      thisPlugin._logger.debug(
+        'MysqlPlugin#patch: patched mysql createConnection'
+      );
+
+      return function createConnection(
+        _connectionUri: string | mysqlTypes.ConnectionConfig
+      ) {
+        //@ts-ignore
+        const originalResult = originalCreateConnection(...arguments);
+
+        shimmer.wrap(
+          originalResult,
+          'query',
+          thisPlugin._patchQuery(originalResult) as any
+        );
+
+        return originalResult;
+      };
+    };
+  }
+
+  // global export function
+  private _patchCreatePool() {
+    return (originalCreatePool: typeof mysqlTypes.createPool) => {
+      const thisPlugin = this;
+      thisPlugin._logger.debug('MysqlPlugin#patch: patched mysql createPool');
+      return function createPool(_config: string | mysqlTypes.PoolConfig) {
+        //@ts-ignore
+        const pool = originalCreatePool(...arguments);
+
+        shimmer.wrap(pool, 'query', thisPlugin._patchQuery(pool));
+        shimmer.wrap(
+          pool,
+          'getConnection',
+          thisPlugin._patchGetConnection(pool)
+        );
+
+        return pool;
+      };
+    };
+  }
+
+  // global export function
+  private _patchCreatePoolCluster() {
+    return (originalCreatePoolCluster: typeof mysqlTypes.createPoolCluster) => {
+      const thisPlugin = this;
+      thisPlugin._logger.debug(
+        'MysqlPlugin#patch: patched mysql createPoolCluster'
+      );
+      return function createPool(_config: string | mysqlTypes.PoolConfig) {
+        //@ts-ignore
+        const cluster = originalCreatePoolCluster(...arguments);
```

Comment:
Could we avoid using `@ts-ignore` there by using `originalCreatePoolCluster.apply(this, arguments)` ?

Reply:
Did not want to override the original `this` binding, so I removed the `@ts-ignore` in a different way. PTAL https://github.com/dynatrace-oss-contrib/opentelemetry-js/commit/031e60b95b52cb1aaa7628c34f93f782bd44d5d9